backend/calc.py

```diff
@@ -68,34 +114,45 @@ def __init__(self, job, wd, mainstat, det, crit, dh, speed, ten, pie):
         self.pie = Stat(Stats.PIE, pie)
 
     @classmethod
-    def truncate(cls, val, precision=1000):
+    def truncate(cls, val: float, precision=1000) -> float:
         return (precision + val) / precision
 
     @classmethod
-    def multiply_and_truncate(cls, val, factor, precision=1000):
+    def multiply_and_truncate(cls, val: float, factor: float, precision=1000):
         return math.floor(val * cls.truncate(factor, precision))
 
     @classmethod
-    def apply_stat(cls, damage, stat):
+    def apply_stat(cls, damage: float, stat: Stat):
         return cls.multiply_and_truncate(damage, stat.get_multiplier())
 
-    # comp is a Comp() object
-    def calc_damage(self, potency, comp, is_dot=False, crit_rate=None, dh_rate=None):
+    def calc_damage(self, potency: int, comp: Comp, is_dot=False, crit_rate=None, dh_rate=None) -> float:
+        """
+        Calculates the estimated DPS based on the team composition and current character stats
+        :param potency: Potency calculated on expected rotation
+        :param comp: Team composition.
+        :param is_dot: ???
+        :param crit_rate: ???
```

Comment:
manual override for crit/dh rates as opposed to rates based on gear stats. will be used in the future to provide stats based on nth percentile crit/dh luck for people who wanna grind for that sweet sweet 99th percentile crit luck run

Reply:
noted and added